jekyll/_cci2/language-java.md

```diff
@@ -84,73 +117,108 @@ version: 2
 
 Next, we have a `jobs` key. Each job represents a phase in your Build-Test-Deploy process. Our sample app only needs a `build` job, so everything else is going to live under that key.
 
-In each job, we must specify a `working_directory`. In this sample config, we’ll name it after the project in our home directory.
-
 ```yaml
 version: 2
 jobs:
   build:
-    working_directory: ~/circleci-demo-java-spring
+    # Remove if parallelism is not desired
+    parallelism: 2
+    environment:
+      # Configure the JVM and Gradle to avoid OOM errors
+      _JAVA_OPTIONS: "-Xmx3g"
+      GRADLE_OPTS: "-Dorg.gradle.daemon=false -Dorg.gradle.workers.max=2"
 ```
 
-This path will be used as the default working directory for the rest of the `job` unless otherwise specified.
+An optional `parallelism` value of 2 is specified as we would like to run tests in [parallel](https://circleci.com/docs/2.0/parallelism-faster-jobs/) to speed up the job.
 
-Directly beneath `working_directory`, we can specify container images under a `docker` key.
+We also use the `environment` key to configure the JVM and Gradle to [avoid OOM errors](https://circleci.com/blog/how-to-handle-java-oom-errors/).
```

Comment:
Excellent, thank you for linking to this blog entry. Definitely something we find ourselves pointing users to from time to time.